File: frontend/src/api/strategy.ts
// 与 backend/app/schemas/strategy.py 对齐
import client from './client'

export interface ParamDef {
  name: string
  type: string
  default: unknown
  annotation: string | null
  required: boolean
}

export interface StrategySummary {
  name: string
  module: string
  file: string
  description: string
  params_count: number
  live_status: string
}

export interface StrategyDetail {
  name: string
  module: string
  file: string
  description: string
  class_docstring: string
  params: ParamDef[]
  signal_kind: string
  runtime_params: string[]
  live_status: string
}

export interface StrategyListResponse {
  strategies: StrategySummary[]
  total: number
}

export async function fetchStrategyList(): Promise<StrategyListResponse> {
  return client.get<unknown, StrategyListResponse>('/strategy')
}

export async function fetchStrategyDetail(name: string): Promise<StrategyDetail> {
  return client.get<unknown, StrategyDetail>(`/strategy/${encodeURIComponent(name)}`)
}

File: backend/app/schemas/strategy.py
from typing import Any

from pydantic import BaseModel


class ParamDef(BaseModel):
    name: str
    type: str
    default: Any = None
    annotation: str | None = None
    required: bool


class StrategySummary(BaseModel):
    name: str
    module: str
    file: str
    description: str
    params_count: int
    live_status: str


class StrategyDetail(BaseModel):
    name: str
    module: str
    file: str
    description: str
    class_docstring: str
    params: list[ParamDef]
    signal_kind: str
    runtime_params: list[str]
    live_status: str


class StrategyListResponse(BaseModel):
    strategies: list[StrategySummary]
    total: int
